File: reactapp/src/services/reminderService.js
import apiClient from './apiClient';

export const createReminder = (userId, documentId, message, remindAt) =>
  apiClient.post('/api/reminders', { userId, documentId, message, remindAt });

export const getRemindersByUser = (userId) =>
  apiClient.get(`/api/reminders/user/${userId}`);

export const getRemindersByDocument = (documentId) =>
  apiClient.get(`/api/reminders/document/${documentId}`);

export const dismissReminder = (reminderId) =>
  apiClient.patch(`/api/reminders/${reminderId}/dismiss`);

export const deleteReminder = (reminderId) =>
  apiClient.delete(`/api/reminders/${reminderId}`);

File: springapp/src/main/java/com/examly/springapp/service/ReminderService.java
package com.examly.springapp.service;

import com.examly.springapp.model.Reminder;
import java.time.LocalDate;
import java.util.List;

public interface ReminderService {
    Reminder createReminder(Long documentId, String documentName, Long userId, LocalDate reminderDate, int daysBefore, String description);
    /** Simplified create used by ReminderController */
    Reminder createReminder(Long userId, Long documentId, String message, String remindAt);
    List<Reminder> getReminders(Long userId);
    /** Alias used by ReminderController */
    List<Reminder> getRemindersByUser(Long userId);
    /** Used by ReminderController */
    List<Reminder> getRemindersByDocument(Long documentId);
    List<Reminder> getActiveReminders(Long userId);
    /** Returns updated Reminder entity */
    Reminder dismissReminder(Long reminderId);
    void deleteReminder(Long reminderId);
    List<Reminder> getDueReminders();
    int processDueReminders();
}
